Add 0.5.0 release announcement blog (#43)
kroxylicious/kroxylicious.github.io: _posts/2024-03-11-release-0_5_0.md

---
layout: post
title:  "Kroxylicious release 0.5.0"

> date:   2024-03-10 00:00:00 +0000

Author: todo get the date sorted, it needs to be in the past for jekyll to present it for testing.

author: "Robert Young"
author_url: "https://www.github.com/robobario"
categories:  [releases, kroxylicious, record-encryption]
---

The Kroxylicious project is very pleased to announce the release of [Kroxylicious 0.5.0](https://github.com/kroxylicious/kroxylicious/releases/tag/v0.5.0).

### Record Encryption

> This release was focused on refining [Record Encryption](/use-cases) to be ready for experimentation in secure environments, adding:

Reviewer: Grr annoying we can't link directly to the use-case it self.

Reviewer 2: Sure you can, it would be `/use-cases/#encryption-at-rest`, i.e.
```suggestion
This release was focused on refining [Record Encryption](/use-cases/#encryption-at-rest) to be ready for experimentation in secure environments, adding:
```

Reviewer: I didn't think we could link directly to catalog entries, is that only true for tabbed ones?

* Enable users to configure Hashicorp Vault TLS, providing custom keystore and truststore
* Enable users to supply the Hashicorp Vault token via file rather than having to inject it into the proxy configuration
* Include Record Encryption filter (and other Kroxylicious project supported filters) in the binary distribution
* Release a Docker image `quay.io/kroxylicious/kroxylicious` also containing the supported filters

> * Internal refactoring to control the usage of key material, preparing to support alternate Ciphers and configurable Additional Authenticated Data

Reviewer: Not entirely sure we want to talk about refactoring per se but focus on what it means. How about? 
```suggestion
* Improvements in how we manage and control the usage of key material
* preparing to support alternate Ciphers and configurable Additional Authenticated Data
```

> The protocol for immutable encrypted data written to the broker is now aligned with our initial design and we guarantee data encrypted with version 0.5.0 
> of the Filter will be decryptable forever (assuming the keys stored in the KMS are available forever for decryption).

Reviewer 2: Maybe reword this slightly?
```suggestion
The protocol for immutable encrypted data written to the broker is now aligned with our initial design and we guarantee data encrypted with version 0.5.0 
of the Filter will be decryptable by all future versions of Kroxylicious (assuming the encryption keys stored in the KMS remain available).
```
I feel like this makes it a bit more clear that we don't mean decryptable by *anything*, we specifically mean future versions of Kroxylicious. CYA, etc.

Author: Done, though I've changed to `future versions of the Record Encryption Filter`, I figured they might evolve separately. WDYT? Too confusing to make the distinction here?

Reviewer 2: That makes sense to me 👍

> > **_NOTE:_**: Until now we have used the term Envelope Encryption when describing the Filter. Going forward we will use the name Record Encryption 
> to better describe the granularity that encryption is happening at. Encryption is done per [record](https://kafka.apache.org/documentation/#record).
> Envelope Encryption is how we encrypt those records. Record Encryption is how we implement Encryption-At-Rest.

Reviewer 2: This could possibly go in a [bootstrap alert element](https://getbootstrap.com/docs/5.3/components/alerts/), we have a pre-made one you can include at [`_includes/bs-alert.html`](https://github.com/kroxylicious/kroxylicious.github.io/blob/main/_includes/bs-alert.html) (there are examples of it being used in [`_quickstarts/deploy.markdown`](https://github.com/kroxylicious/kroxylicious.github.io/blob/main/_quickstarts/deploy.markdown?plain=1#L24)). I think it looks fine as it is, but using the `bs-alert` element might make it a bit more uniform with the quickstarts.

For example:
```suggestion
{% capture record_encryption_note %}
Until now we have used the term Envelope Encryption when describing the Filter. Going forward we will use the name Record Encryption 
to better describe the granularity that encryption is happening at. Encryption is done per [record](https://kafka.apache.org/documentation/#record).
Envelope Encryption is how we encrypt those records. Record Encryption is how we implement Encryption-At-Rest.
{% endcapture %}

{% include bs-alert.html type="primary" icon="info-circle-fill" content=record_encryption_note %}
```

Which looks a bit like this:
![image](https://github.com/kroxylicious/kroxylicious.github.io/assets/15976368/7cb218d7-b6bd-4661-829a-69137e5bd108)

Author: Have updated to use the banner

### Other Improvements

> * Support for Apache Kafka 3.7.0 API additions (Kroxylicious is forwards-compatible, but must be modified if we wish to intercept new
> APIs or fields added to existing APIs)

Reviewer: I think we want to expand this a little.

Something like 
```suggestion
* Support for Apache Kafka 3.7.0 API additions (Kroxylicious is version agnostic and will automatically interfaces between all broker and client versions, but must be updated to enable the interception of new
APIs or fields added to existing APIs)
```

Reviewer 2: I'm finding the second version harder to follow, personally. I wonder if it could be worded a bit differently, something like
```suggestion
* Support for Apache Kafka 3.7.0 API additions (Kroxylicious is version agnostic and can interface with all current broker and client versions, but must be updated to intercept new
APIs or fields added to existing APIs)
```

Reviewer: What I was trying to express was that you can drop a new client or broker version either side of Kroxylicious and it will just work, there is no need for co-ordinated upgrades. So really it would be `current and future` broker and client versions`. But it does need upgraded to expose or intercept the new APIs.

Though I fully accept that could be far too much nuance for the release notes.

* A fix for upstream TLS connections failing, [@callaertanthony](https://github.com/callaertanthony)'s first PR contribution, thank you!
* Added a `lowestTargetBrokerId` configuration for Port-per-broker Virtual Cluster exposition
* Numerous bugfixes, documentation improvements, test enhancements

### Artefacts

You'll find binaries attached to the GitHub [release](https://github.com/kroxylicious/kroxylicious/releases/tag/v0.5.0), and available through [Maven Central](https://repo1.maven.org/maven2/io/kroxylicious/kroxylicious-app/0.5.0/).

> Docker images are hosted at [quay.io/kroxylicious/kroxylicious](https://quay.io/repository/kroxylicious/kroxylicious)

Reviewer 2: Might be nice to add a line or two at the end here about where people can report bugs and leave feedback

---
layout: post
title:  "Kroxylicious release 0.5.0"
date:   2024-03-10 00:00:00 +0000
author: "Robert Young"
author_url: "https://www.github.com/robobario"
categories:  [releases, kroxylicious, record-encryption]
---

> The Kroxylicious project is very pleased to announce the release of [Kroxylicious 0.5.0](https://github.com/kroxylicious/kroxylicious/releases/tag/v0.5.0). See the [Changelog](https://github.com/kroxylicious/kroxylicious/blob/main/CHANGELOG.md#050) for a list of changes and summary of Deprecations, Changes and Removals.

Author: also added a link to the changelog here
